hrjobcontract/js/src/job-contract/controllers/form/form-leave.js

```diff
@@ -0,0 +1,38 @@
+define([
+    'job-contract/vendor/fraction',
+    'job-contract/controllers/controllers'
+], function (Fraction, controllers) {
+    'use strict';
+
+    controllers.controller('FormLeaveCtrl',['$scope', '$rootScope', '$filter', '$log', '$q', 'UtilsService',
+      function ($scope, $rootScope, $filter, $log, $q, UtilsService) {
+        $log.debug('Controller: FormLeaveCtrl');
+
+        var promiseNumberOfPublicHolidays = UtilsService.getNumberOfPublicHolidaysInCurrentPeriod();
+
+        promiseNumberOfPublicHolidays.then(function(result){
+          var number = parseFloat(result);
```

Comment:
I think that this entire logic here
``js
var number = parseFloat(result);

if (isNaN(number)) {
  number = 0;
}
```
should be in the `getNumberOfPublicHolidaysInCurrentPeriod` method, as it's its responsibility to make sure that a valid value is returned